pkg/deploy/strategy/support/lifecycle.go

```diff
@@ -203,3 +203,121 @@ func NewPodWatch(client kclient.Interface, namespace, name, resourceVersion stri
 		return obj.(*kapi.Pod)
 	}
 }
+
+func NewContainersReadyCanary(kclient kclient.Interface, timeout time.Duration) *ContainersReadyCanary {
```

Comment:
I don't think `ContainersReadyCanary` describes this very well.  I would call the interface PodAcceptor or UpdateAcceptor and describe this as "FirstContainerReady"